ludwig/hyperopt/execution.py

```diff
@@ -806,44 +787,73 @@ def run_experiment_trial(config, local_hyperopt_dict, checkpoint_dir=None):
 
         run_experiment_trial_params = tune.with_parameters(run_experiment_trial, local_hyperopt_dict=hyperopt_dict)
 
+        if _is_ray_backend(backend):
+            # If Ray backend, only request custom resource at trial level (inner Tuner will request resources)
+            resources = [self.trial_function_resources]
+        else:
+            # If not Ray backend, request all of the resources required at the trial level
+            use_gpu = bool(self._gpu_resources_per_trial_non_none)
+            num_cpus, num_gpus = _get_num_cpus_gpus(use_gpu)
+            resources = [self.trial_function_resources, {"CPU": num_cpus, "GPU": num_gpus}]
+
+        resources_per_trial = PlacementGroupFactory(resources)
+        run_experiment_trial_params = tune.with_resources(run_experiment_trial_params, resources_per_trial)
+
         @ray.remote(num_cpus=0)
         def _register(name, trainable):
             register_trainable(name, trainable)
 
         ray.get(_register.remote(f"trainable_func_f{hash_dict(config).decode('ascii')}", run_experiment_trial_params))
 
-        # Note that resume="AUTO" will attempt to resume the experiment if possible, and
-        # otherwise will start a new experiment:
-        # https://docs.ray.io/en/latest/tune/tutorials/tune-stopping.html
-        should_resume = "AUTO" if resume is None else resume
-
         try:
-            analysis = tune.run(
-                f"trainable_func_f{hash_dict(config).decode('ascii')}",
-                name=experiment_name,
-                config={
-                    **self.search_space,
-                    **tune_config,
-                },
-                scheduler=self.scheduler,
-                search_alg=search_alg,
-                num_samples=self.num_samples,
-                keep_checkpoints_num=1,
-                max_failures=1,  # retry a trial failure once
-                resources_per_trial=resources_per_trial,
-                time_budget_s=self.time_budget_s,
-                sync_config=self.sync_config,
-                local_dir=output_directory,
-                metric=metric,
-                mode=mode,
-                trial_name_creator=lambda trial: f"trial_{trial.trial_id}",
-                trial_dirname_creator=lambda trial: f"trial_{trial.trial_id}",
-                callbacks=tune_callbacks,
-                stop=CallbackStopper(callbacks),
-                verbose=hyperopt_log_verbosity,
-                resume=should_resume,
-                log_to_file=True,
-            )
+            tuner_cls = TunerRay210 if not _ray220 else Tuner
+
+            if resume:
+                # Restore from experiment directory
+                tuner = tuner_cls.restore(
```

Comment:
Does this work with our custom syncer?

Reply:
Hm, good question– @arnavgarg1 do we have existing tests for this? I didn't test it explicitly myself.

Reply:
Will test this today, but I believe it should work